src/dataExplorer/context/resultsView.tsx

```diff
@@ -47,19 +65,67 @@ export const ResultsViewProvider: FC = ({children}) => {
     } as SimpleTableViewProperties,
   })
 
-  const [viewOptions, setViewOptions] = useSessionStorage(
+  // what can be chosen (e.g. the list of all options)
+  const [viewOptionsAll, saveViewOptionsAll] = useSessionStorage(
+    'dataExplorer.resultsOptions.all',
+    DEFAULT_VIEW_OPTIONS
+  )
+  const setViewOptionsAll = useCallback(
+    (updatedOptions: Partial<ViewOptions>) => {
+      saveViewOptionsAll({...viewOptionsAll, ...updatedOptions})
+    },
+    [viewOptionsAll]
+  )
+
+  // default options (a.k.a. based on schema)
+  const [defaultViewOptions, saveDefaultViewOptions] = useSessionStorage(
+    'dataExplorer.resultsOptions.default',
+    DEFAULT_VIEW_OPTIONS
+  )
+  const setDefaultViewOptions = useCallback(
```

Comment:
if you choose to name `setViewOptionsAll` as something like `saveViewOptionsAllMemoized`, this should also be updated to follow the same convention

Reply:
I'm hesitant to use the word `memoized`, as in the definition is to remember -- and the ALL options is reset rather frequently (with every parent query run). Depending on the user flow, they could be memoizing the ALL (versus the DEFAULT) more or less often.

Will muse on another naming convention.

Reply:
See if [this last commit added](https://github.com/influxdata/ui/pull/6502/commits/eff939c4937828baa12cc723736fb5a09fed6b36), helps to clarify.

Reply:
Also, here is some manual QA on the behavior for `clear()`.
(Note: we really need to put in tests sometime soon-ish.)


https://user-images.githubusercontent.com/10232835/213325137-697cda14-a337-46ba-8105-eb518176b235.mov

Reply:
Cool, works for me! Thanks for thinking about this critically!